### PR TITLE
Reports

### LINKED ISSUE #5: add attack form
add attack form that should contain

attack start button and a field that allows to alter attack start data time
attack end button and a field that allows to alter attack end data time
add attack abortive treatment button that allows to submit preventive treatment type, doze and start/end of usage

### LINKED ISSUE #6: add attacks table view
add attacks table view

### PR DESCRIPTION
**Pull request by itka4yk**

### CODE REVIEW

File: src/main/java/org/chtracker/web/AttackController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+
+@RestController
+@RequestMapping("/attacks")
+public class AttackController {
+    private final AttackRepository attackRepository;
+
+    @Autowired
```
Comment (madamusinski) [inline src/main/java/org/chtracker/web/AttackController.java]: Unneccesary Annotation. Starting from Spring 4.3 it is ommited to use @Autowired over constructor injector

File: src/main/java/org/chtracker/web/AttackController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @GetMapping
+    public Iterable<Attack> getAttacks(
+            @PageableDefault(sort = "started", direction = Sort.Direction.ASC) Pageable pageable,
+            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime from,
+            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime to
+    ) {
+        // TODO: add filterBy patient
+        if (from == null && to == null) {
```
Comment (madamusinski) [inline src/main/java/org/chtracker/web/AttackController.java]: Logic like this with if ladder is responsibility of service/facade not a controller.

**Review by madamusinski — CHANGES_REQUESTED** [review]

File: src/main/java/org/chtracker/web/AttackController.java (hunk 1 above)
Comment (itka4yk) [reply src/main/java/org/chtracker/web/AttackController.java]: done

File: src/main/java/org/chtracker/web/AttackController.java (hunk 2 above)
Comment (itka4yk) [reply src/main/java/org/chtracker/web/AttackController.java]: done

**Comment by itka4yk** [comment]
Guys, feel free to comment UI/UX side, I'll adjust everything.

image
image

**Comment by itka4yk** [comment]
Create report form with used abortive treatments table.

This is editable table, which create new row after last one was edited.

image
image
image

No validation applied at this moment.

**Pull request closed without merging**